image-rs/src/layer_store.rs

```diff
@@ -0,0 +1,79 @@
+use std::fs;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// The LayerStore abstracts the image layers storage on the file system.
+pub struct LayerStore {
+    /// OCI image layer data store dir.
+    pub data_dir: PathBuf,
+
+    /// Next layer index
+    layers_index: AtomicUsize,
+}
+
+impl LayerStore {
+    /// Retrieves the next layer index by computing the maximum index from the
+    /// existing layers in data dir.
+    pub fn get_layer_index(data_dir: PathBuf) -> AtomicUsize {
+        let mut next = 0;
+        if data_dir.exists() {
+            if let Ok(paths) = fs::read_dir(data_dir) {
```

Comment:
what if this op fails, shouldn't we propagate the error, panic or something else? if we want to swallow it for some reason, we can early exit and maybe log it and save a level of indentation.

```rust
let Ok(paths) = fs::read_dir(data_dir) else {
   // maybe log?
   return AtomicUsize::new(0);
}
```

Reply:
Added logging.

Reply:
I changed to propagate the error